File: github/src/main/webapp/js/hello.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

function onReady() {
        
        // 締日の自動入力
        $("#datadate").on("change", function () {
            
           // 起票日を取得
           var simeDate = $("#datadate").val();
           
           // 日付セパレータでsplit
           var simeSplit = simeDate.split("-");
           
           // 3個に出来たら
           if ( simeSplit.length === 3 ) {
               var sDate = new Date(simeSplit[0], simeSplit[1], simeSplit[2]);
               
               // 日付として認可出来るなら
               if ( sDate.getDate() > 0 && simeSplit[2].match(/[0-9]+/) ) {
                   // 締日を計算する
                   var monp = 1;
                   
                   // 9日以上なら翌々月
                   if ( parseInt(simeSplit[2]) > 8 ) {
                       monp = 2;
                   }
                   
                   // 付きの0サブ
                   simeSplit[1] = parseInt(simeSplit[1]) + monp ;
                   if ( parseInt(simeSplit[1]) > 12 ) {
                       simeSplit[0] ++;
                       simeSplit[1] = parseInt(simeSplit[1]) - 12;
                   }
                   simeSplit[1] = "0" + simeSplit[1] ;
                   simeSplit[1] = simeSplit[1].substring(simeSplit[1].length - 2);
                   simeSplit[2] = '05';
                   
                   // 締日を自動で入力
                   $("#sime").val( simeSplit[0] + "-" + simeSplit[1] + "-" + simeSplit[2] );
               }
           }
        }
        );
    
}

File: github/src/main/java/com/oonoyoshihito/spark/oono/Hello.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.oonoyoshihito.spark.oono;

import com.oonoyoshihito.spark.velocity.AppBase;
import com.oonoyoshihito.spark.velocity.HtmlTableHelper;
import java.util.HashMap;
import java.util.Map;
import spark.ModelAndView;
import spark.Request;
import spark.Response;
import spark.Session;

/**
 *
 * @author duke_2
 */
public class Hello extends GetRequest {

    @Override
    public ModelAndView doGet(Request request, Response response) {
        HtmlTableHelper hth = AppBase.getHtmlTable("select * from siharai order by id desc", false);
        String htmlTable = AppBase.getSiharaiSum() + "<br/>" + hth.makeTableInner() + "<div>" + request.contextPath() + "</div>";
        Map<String, Object> model = new HashMap<>();
        model.put("table", htmlTable);
        model.put("result", hth.core.size());
        model.put("root", request.contextPath());
        Session s = request.session();
        if ( s != null ) {
            model.put("error", s.attribute("error"));
        } else {
            model.put("error", " ");
        }

        // The wm files are located under the resources directory
        return new ModelAndView(model, "hello.wm");
    }
}
